packages/grafana-ui/src/components/PieChart/PieChartWithLegend.tsx

```diff
@@ -1,8 +1,29 @@
 import React, { FC } from 'react';
-import { Props as PieChartProps } from './PieChart';
+import { VizLayout } from '../VizLayout/VizLayout';
+import { LegendDisplayMode, VizLegendItem } from '../VizLegend/types';
+import { VizLegend } from '../VizLegend/VizLegend';
+import { PieChart, DEFAULT_COLOR, Props as PieChartProps } from './PieChart';
 
 export interface Props extends PieChartProps {}
 
-export const PieChartWithLegend: FC<Props> = ({ width, height, ...restProps }) => {
-  return <div>Need VizLayout in grafana/ui</div>;
+export const PieChartWithLegend: FC<Props> = ({ values, width, height, ...restProps }) => {
+  const legendItems = values.map<VizLegendItem>((value) => {
+    return {
+      label: value.title ?? '',
+      color: value.color ?? DEFAULT_COLOR,
+      yAxis: 1,
+    };
+  });
+
+  const legendElement = (
+    <VizLegend items={legendItems} placement="right" displayMode={LegendDisplayMode.List}></VizLegend>
+  );
+
+  return (
+    <VizLayout width={width} height={height} legend={legendElement}>
+      {(vizWidth: number, vizHeight: number) => (
+        <PieChart values={values} width={vizWidth} height={vizHeight} {...restProps}></PieChart>
```

Comment:
Maybe change to this:
```suggestion
        <PieChart values={values} width={vizWidth} height={vizHeight} {...restProps} />
```